python/lsst/ts/linearstage/controllers/zaber_lst.py

```diff
@@ -19,23 +19,244 @@
+        """Connect to the device."""
+        if self.simulation_mode:
+            self.mock_server = LinearStageServer(port=0, log=self.log)
+            await self.mock_server.start_task
+            self.config.port = self.mock_server.port
+        try:
+            self.client = await Connection.open_tcp_async(
+                host_name=self.config.hostname, port=self.config.port
+            )
+        except ConnectionFailedException:
+            self.log.exception("Failed to connect to host/port.")
+            raise RuntimeError(f"Unable to connect to host {self.config.hostname}.")
+        self.device = self.client.get_device(self.config.daisy_chain_address)
+        self.log.debug(f"{self.device=}")
+        await self.device.identify_async()
+        self.log.debug(f"{self.device=}")
+        self._check_axes()
+        await self.update()
+
+    async def disconnect(self) -> None:
+        """Disconnect from the device."""
+        if self.client is not None:
+            self.client.close()
+        self.device = None
+        self.axes = []
+        self.client = None
+        if self.simulation_mode:
+            if self.mock_server is not None:
+                await self.mock_server.close()
+            self.mock_server = None
+
+    async def move_relative(self, value: float) -> None:
+        """Move device to position with relative target.
+
+        Parameters
+        ----------
+        value : `float`
+            The amount to move by.
+        """
+        if self.device is None:
+            raise RuntimeError("Device has not been received.")
+        for axis in self.axes:
+            # FIXME DM-45063 Replace with _perform when parameters are
+            # supported.
+            try:
+                await axis.move_relative_async(
+                    position=value, unit=Units.LENGTH_MILLIMETRES
+                )
+            except CommandFailedException as cfe:
+                self.log.exception(f"Move relative rejected for {cfe.message}")
+                raise
+
+    async def move_absolute(self, value: float) -> None:
+        """Move the device to value.
+
+        Parameters
+        ----------
+        value : `float`
+            The position to move to.
+        """
+        if self.device is None:
+            raise RuntimeError("Device has not been received.")
+        for axis in self.axes:
+            # FIXME DM-45063 Replace with _perform when parameters are
+            # supported.
+            try:
+                await axis.move_absolute_async(
+                    position=value, unit=Units.LENGTH_MILLIMETRES
+                )
+            except CommandFailedException as cfe:
+                self.log.exception("Move absolute rejected.")
+                raise salobj.ExpectedError(f"Move absolute rejected for {cfe.message}")
+
+    async def home(self) -> None:
+        """Home the device, needed to gain awareness of position."""
+        if self.device is None:
```

Comment:
Please add the whole if and for loop as well like this:
```
        if self.device is None:
            raise RuntimeError("Device has not been received.")
        for axis in self.axes:
	        try:
	            await command()
	        except CommandFailedException:
	            self.log.exception(f"{command} rejected.")
	            raise
```

Reply:
I don't think that's going to work with passing the axis method directly.